packages/integrations/cloudflare/src/index.ts

```diff
@@ -68,14 +68,29 @@ export default function createIntegration(args?: Options): AstroIntegration {
 	return {
 		name: '@astrojs/cloudflare',
 		hooks: {
-			'astro:config:setup': ({ config, updateConfig }) => {
+			'astro:config:setup': ({ config, updateConfig, logger }) => {
+				let imageServiceConfig = config.image.service;
+
+				// If the user is using Squoosh or Sharp, let's warn and tell them we're instead using the noop service
+				if (
+					['astro/assets/services/squoosh', 'astro/assets/services/sharp'].includes(
+						config.image.service.entrypoint
+					)
+				) {
+					logger.warn(
+						`The currently selected image service ${config.image.service.entrypoint} is not compatible with the current adapter. In order to avoid breaking your project, an alternative image service that does not process images has been applied. To suppress this warning, manually specify \`image: {service: { entrypoint: 'astro/assets/services/noop'}}\` in your project's config.`
+					);
+					imageServiceConfig = { entrypoint: 'astro/assets/services/noop', config: {} };
+				}
```

Comment:
The problem is that the validation outputs an invalid error message if the config update isn't done before `configDone`, I wasn't sure how to handle this without having to make a one off for this, so I figured doing it in the adapter was a better solution, but yeah, it's not amazing

Reply:
I see. Maybe we can update the invalid error message to point using noop instead? About automatically switching to the noop, maybe it's fine to not do that since anyone would've want to get rid of the warnings/errors anyways.

Ideally it would be nice if the config can fallback to noop automatically without warning, but only warn if the user explicit sets that, but checking the flow it's not easy to do that.

Maybe @ematipico has ideas about this?

Reply:
I think we can change the logic of the validation and messaging to meet our expectations. Also, I think it's fine to apply the change in configuration here: 

https://github.com/withastro/astro/blob/25b89359e90dac3a79c4282f07901e76133b71fa/packages/astro/src/integrations/index.ts#L225

If the adapter is not compatible with either service, we hardcode the noop service.